Tasks/Common/azure-arm-rest/webRequestUtility.ts

```diff
@@ -0,0 +1,23 @@
+import tl = require("vsts-task-lib/task");
+import webClient = require("./webClient");
+const HttpRedirectCodes: number[] = [301, 302, 307, 308];
+class WebRequestUtility {
+    public static async getTargetUriFromFwdLink(fwdLink: string) {
+        tl.debug("Trying to fetch target link from the fwdlink: " + fwdLink);
+        var httpRequest = new webClient.WebRequest();
+        httpRequest.method = 'GET';
+        httpRequest.uri = fwdLink;
+        var httpResponse = await webClient.sendRequest(httpRequest);
+        if(HttpRedirectCodes.indexOf(httpResponse.statusCode) == -1) {
+            throw new Error(`HttpResponse statuscode: ${httpResponse.statusCode} is not a valid HTTP redirect code.`); 
```

Comment:
If it's not a redirect and status code is 200, why don't you send the actual URI only instead of failing. For idempotency

Reply:
if the typed rest client version is updated, then a 'GET' on fwdlink would start auto-redirecting and giving back the target resource instead of the desired 3xx response. And if that happens, I would want this function to throw so that the integration tests can catch that and we can fix the initialization of webclient accordingly( so that it does not auto-redirect). And this is because if the agent is behind proxy, it may have access to aka.ms but not to the actual target resource. In that case, the call with auto-redirect will fail. We do not actually need that resource, but just the url.

Reply:
Please ensure there is test coverage so that any change to pick latest webclient (which supports AutoRedirect by default) does not regress this scenario.